docs/approach-docs/platform-containers.md

```diff
@@ -0,0 +1,98 @@
+# A Containerized Approach Using Docker and Compose - Approach Doc
+
+I would like to propose the adoption of docker and docker-compose to begin containerizing the Journaly stack, starting with the database, paving the way for other servers or services that require installing software on a local development machine.
+
+## Details
+
+Installing postgres directly on a development machine might not be possible for some, and requires a level of support and maintenance on their part. With a containerized approach, we can eliminate this hurdle and perhaps shorten the time it takes for new developers to get setup. 
+
+Additionally, we could have a more consistent database environment for all developers, for instance, setting up users and other related artifacts during the containerization process.
+
+When I first setup Journaly, I used docker for my DB, and with the following changes, had it up and running in little time.
+
+### Docker
+
+We need a root level `Dockerfile` that looks something like this:
+
+```docker
+FROM postgres:11
+
+EXPOSE 5432
+
+COPY bootstrap.sh /docker-entrypoint-initdb.d/db-bootstrap.sh
+```
+
+You'll see that I have a bootstrap shell script to set up some initial database work:
+
+```sh
+#!/bin/bash
+
+set -e # bail immediately on error
+
+psql <<- EOSQL
+    CREATE USER developer WITH CREATEDB PASSWORD 'nights_watch';
+    CREATE DATABASE journaly_db;
+EOSQL
+
+echo "Database bootstrap complete."
+```
+
+There's a lot of potential here to automate environment and database settings.
+
+### Docker-Compose
+
+Finally, we need a `docker-compose.yml` file to tie it all together:
+
+```yaml
+version: '3'
+
+services:
+  journaly:
+    image: 'journaly:latest'
+    ports:
+      - '5432:5432'
+    volumes:
+      - journaly_db:/var/lib/postgresql/data
+    environment:
+      POSTGRES_PASSWORD: <db_password>
+    build: .
+
+volumes:
+  journaly_db:
+
+```
+
+### Final Modifications
+
+With the given files, we end up with the following additions:
+
+- `db-bootstrap.sh`
+- `Dockerfile`
+- `docker-compose.yaml`
```

Comment:
The proposal will dockerize the DB but not the actual app process, e.g. `npm run dev`. I'd like to add another container that will dockerize the actual development server. The benefits there would be:

- `docker-compose up` will bring up both the DB and the dev server
-  we won't have to fuss with cross platform support for prisma and nexus. Also npm scrips use cross-env and we could probably get away with kicking that dep if we could say "dockerized env is the supported dev setup on windows"
- we could run some of the one off stuff like `prisma generate` and nexus reflection as either part of the build or startup phase, so just that many less hoops for developers to jump though

We could do the work in two passes, just the DB then the later the application itself, but I think there are some some definite wins in doing both

Reply:
I agree with this! In this case, I am wondering if I should try to tackle the project re-org first then? I was going to try and do it in the next day or two and see how seamless it ends up being? #400

Reply:
Yeah, re-orging first would probably make our lives simpler. "web" (or whatever we call the main thing) having its own dir instead of just being the project root would make sense as a place for a docker file, and the compose file could live at the actual root.